library/core/src/mem/maybe_uninit.rs

```diff
@@ -810,8 +812,9 @@ impl<T> MaybeUninit<T> {
     ///
     /// [`assume_init_ref`]: MaybeUninit::assume_init_ref
     #[unstable(feature = "maybe_uninit_slice", issue = "63569")]
+    #[rustc_const_unstable(feature = "const_maybe_uninit_assume_init", issue = "none")]
     #[inline(always)]
-    pub unsafe fn slice_assume_init_ref(slice: &[Self]) -> &[T] {
+    pub const unsafe fn slice_assume_init_ref(slice: &[Self]) -> &[T] {
```

Comment:
What is the reason that you made the slice methods `const` but not `assume_init_ref` and `assume_init_mut`?

Reply:
I believe I skipped it since I got errors about `&*self.value` doing not-const things, guess it might be about `self.value` being `ManuallyDrop<T>` and thus `ManuallyDrop::deref` being a trait method?

Would this
```rust
&*(&self.value as *const _ as *const T)
```
be a valid const equivalent? It does pass `./x.py check` however I am not that confident in my `unsafe` skills so not sure if it is a good solution. Currently running `./x.py test` locally...

Reply:
`&*self.as_ptr()` seems good, if that passes in a `const fn`. :)